sgx/demo/src/util/demo_tls_util.c

```diff
@@ -249,20 +263,49 @@ int demo_tls_config_client_connect(TLSPeer * tls_clnt)
     return -1;
   }
 
-  // set hostname for Server Name Indication
-  if (1 != SSL_set_tlsext_host_name(ssl, tls_clnt->host))
+  // compute expected server name (as specified in server's cert)
+  //   - expected to be <IP address or hostname string>.<name string>
+  //     (e.g., localhost.demoServer, 127.0.0.1.server, ...)
+  char * server_name = NULL;
+  unsigned int server_name_size = strlen(tls_clnt->remote_server) + 1;
+  if (tls_clnt->remote_server_func != NULL)
+  {
+    server_name_size += (strlen(tls_clnt->remote_server_func) + 1);
+  }
+  server_name = calloc(server_name_size, sizeof(char));
+  int bytes_needed = 0;
+  if (tls_clnt->remote_server_func == NULL)
+  {
+    bytes_needed = snprintf(server_name,
+                            server_name_size,
+                            "%s",
+                            tls_clnt->remote_server);
+  }
+  else
+  {
+    bytes_needed = snprintf(server_name,
+                            server_name_size,
+                            "%s.%s",
+                            tls_clnt->remote_server,
+                            tls_clnt->remote_server_func);
+  }
+  if (bytes_needed > server_name_size)
```

Comment:
Should we also check ```snprintf``` for negative output (indicating failure?)

Reply:
Yes, we should.

The intent of commit [ccf860f](https://github.com/NationalSecurityAgency/kmyth/pull/189/commits/ccf860f0789470f3ab127eea14b3eafb035f7b10) is to add this check.